### PR TITLE
Change BLS Public Key Length Check in Beacon RPC Server

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #1565 

---

# Description

**Write why you are making the changes in this pull request**

Currently, our beacon RPC server checks if the public key in the assignments request is of length 48, however, our BLS library created by Herumi has pubkeys of length 96. Until we find a better BLS library or roll out our own, we will be stuck with a crashing runtime due to this strict check.

**Write a summary of the changes you are making**

This changes the length to global config param which can be checked against instead of a magic number.

### CODE REVIEW

File: beacon-chain/rpc/validator_server.go [hunk 1]
```diff
@@ -43,8 +43,8 @@ func (vs *ValidatorServer) ValidatorEpochAssignments(
 	ctx context.Context,
 	req *pb.ValidatorEpochAssignmentsRequest,
 ) (*pb.ValidatorEpochAssignmentsResponse, error) {
-	if len(req.PublicKey) != 48 {
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator_server.go]: Why change this? Didn't we have panics where invalid public keys were sent?

Comment (prestonvanloon) [reply beacon-chain/rpc/validator_server.go]: IMO, this needs to be fixed to the expectation size. If it's 96 now then it should be 96.

File: beacon-chain/rpc/validator_server_test.go [hunk 2]
```diff
@@ -171,7 +188,7 @@ func TestValidatorCommitteeAtSlot_CrosslinkCommitteesFailure(t *testing.T) {
 	}
 }
 
-func TestValidatorCommitteeAtSlot_Ok(t *testing.T) {
+func TestValidatorEpochAssignments_CorrectAssign(t *testing.T) {
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator_server_test.go]: CorrectAssign? Can you make this human readable?

File: beacon-chain/rpc/validator_server.go (hunk 1 above)
Comment (terenc3t) [reply beacon-chain/rpc/validator_server.go]: pub key size should really become its own variable if it's referenced in multiple places

Comment (prestonvanloon) [reply beacon-chain/rpc/validator_server.go]: yes, it can be a params constant

**Comment by nisdas** [comment]
@rauljordan This would clash with the spec though, a lot of the data structures take the public keys as 48 bytes, cant we just lop off half the bits to create a public key with 48 bytes as a temporary solution ?

**Comment by terenc3t** [comment]
From eth2.0 spec's point of view, it doesn't care for the size of the pub key, it's all abstracted out. As long this doesn't break `bls_verify` (which I don't think it should) then we are good

**Comment by nisdas** [comment]
@terenc3t data structures like this https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#depositinput , and also https://github.com/ethereum/eth2.0-specs/blob/master/specs/bls_signature.md#bls_verify in the bls spec assume 48 bytes. Also the `deposit_data` argument for the deposit function in the deposit contract is hardcoded to expect a pubkey of 48 bytes not 96 bytes. Its not really abstracted away; we could change all this to 96 bytes, but why not just cut off the last 48 bytes of the pub key ? That way we are still in sync with the spec

**Comment by terenc3t** [comment]
I see. That makes sense for the time being but we won't be compatible with rest of the clients :P

**Comment by rauljordan** [comment]
In my opinion, once we implement our own BLS we'll have 48byte pubkeys, so this won't be an issue. The cutoff thing sounds a bit weird to do, so I'd rather just stick to this for now until we have our BLS that is in line with the spec.

**Pull request merged**